### DISCUSSION TITLE
Warning: Expected server HTML to contain a matching <button> in <div>

### DISCUSSION THREAD

**Discussion by paresiqbal** [vercel/next.js#47163] [Help]
### Summary

Hi i need help at this, i created navbar and nextjs using react-cookies to set Logout button after Login. i already use "use client" on the top of the code.
here is the relevant code : 
```
export default function Navbar() {
  const [cookies, setCookies] = useCookies(["access_token"]);
  };

  return (
    <div>
      {!cookies.access_token ? (
        <Link href="/auth/Login">Login</Link>
      ) : (
        <button>Logout</button>
      )}
    </div>
  );
}
```
And this is what the warning give it to me : 
```
Warning: Expected server HTML to contain a matching <button> in <div>.
    at button
    at div
    at Navbar 
    at body
    at html
    at ReactDevOverlay
```
i already try few solution using useEffect base on this https://github.com/vercel/next.js/discussions/17443, like this:
```
export default function Navbar() {
  const [cookies, setCookies] = useCookies(["access_token"]);

 const isLogin = () => {
    cookies.access_token;
  };

 useEffect(() => {
    isLogin();
  }, []);

  return (
    <div>
      {!isLogin ? (
        <Link href="/auth/Login">Login</Link>
      ) : (
        <button>Logout</button>
      )}
    </div>
  );
}
```
But it seems doesn't work, what do i do wrong ? any kind of help will be appreciated, and i am beginer at web development.
Thank you for your help

**Comment by icyJoseph** [comment]
Hi,

I think you need to review a few JavaScript concepts, such as closures, truthy/falsy, and implicit return from arrow functions.

For example, in this piece of code:

```jsx
export default function Navbar() {
  const [cookies, setCookies] = useCookies(["access_token"]);

 // this function returns undefined
 const isLogin = () => {
    cookies.access_token; // should be: return cookies.access_token
  };

 useEffect(() => {
    isLogin(); // this does nothing, no state is set, and the return value is not used at all.
  }, []);

  return (
    <div>
      {/* isLogin is a function, truthy, and here you do not call it, like isLogin(), instead you are casting it to boolean*/}
      {!isLogin ? ( // so this is like saying !true, so it evaluates to false
        <Link href="/auth/Login">Login</Link>
      ) : (
        <button>Logout</button>
      )}
    </div>
  );
}
```

I've commented a few things, please do review that. Additionally, I'd say that working with negations is not very helpful, especially when there's issues, isn't it better to write:

```jsx
{isLoggedIn ? <button>Logout</button> :  <Link href="/auth/Login">Login</Link>}
```

Onto your issue. If the error message says that it expected a button (the logout button I presume), then I have to believe that SSR sees a logged in user, but on the client, the hydration process sees a logged out user.

Is the access_token cookie, `httpOnly`?

**Reply by paresiqbal** [reply]
I have tried what you have suggested, and it work Thank you very much sir

**Reply by icyJoseph** [reply]
What do you mean with "removing the access_token"? How exactly are you doing that?
